core/src/main/scala/dogs/BiPartition.scala

```diff
@@ -0,0 +1,36 @@
+/**
+  * Created by anicolaspp on 4/10/16.
+  */
+
+package dogs
+
+import Predef._
+import simulacrum.typeclass
+import scala.{inline,Iterable}
+import java.lang.{String,StringBuilder}
+import scala.annotation.{tailrec}
+import dogs.syntax.birds._
+import cats._
+
+
+@typeclass trait BiPartition[A] {
+  def partition(predicate: A => Boolean): (List[A], List[A])
```

Comment:
Alternative is the thing that is both a Applicative (giving us pure, a way to go from A => F[A]), and MonoidK, which is a higher kinded monoid, giving us empty: () => F[A] and append: F[A] => F[A] => F[A].  This is the minimum you need from the cats typeclasses to get something like a builder. Because from these three functions. for example, you can do:

    def prepend[F[_]](fa: F[A])(a: A)(implicit F: Alternative[F]) = F.append(F.pure(a), fa)
    def append[F[_]](fa: F[A])(a: A)(implicit F: Alternative[F]) = F.append(fa, F.pure(a))